src/java.base/share/classes/java/time/Duration.java

```diff
@@ -1429,12 +1429,9 @@ public boolean equals(Object otherDuration) {
         if (this == otherDuration) {
             return true;
         }
-        if (otherDuration instanceof Duration) {
-            Duration other = (Duration) otherDuration;
-            return this.seconds == other.seconds &&
-                   this.nanos == other.nanos;
-        }
-        return false;
+        return (otherDuration instanceof Duration other)
+                && this.seconds == other.seconds
+                && this.nanos == other.nanos;
     }
```

Comment:
Perhaps rename the argument and use `otherDuration` as the refinement.
Otherwise, an inconsistency across various classes will creep in where the more specific variable has a more general name.  In this case, the argument type is Object, so the argument name `otherDuration` is not strictly true.

Reply:
Parameter and pattern variable names swapped, as suggested. See 647bd6b